src/controller/key.rs

```diff
@@ -25,27 +27,62 @@ pub enum Key {
 }
 
 impl Key {
-    pub fn from_keycode(keycode: KeyCode) -> Key {
-        match keycode {
-            KeyCode::Char('q') => Key::Quit,
-            KeyCode::Esc => Key::Escape,
-            KeyCode::Enter => Key::Enter,
-            KeyCode::Char('r') => Key::Reload,
-            KeyCode::Char('h') => Key::Left,
-            KeyCode::Char('l') => Key::Right,
-            KeyCode::Char('k') => Key::Up,
-            KeyCode::Char('j') => Key::Down,
-            KeyCode::Char('i') => Key::Insert,
-            KeyCode::Char('0') => Key::About,
-            KeyCode::Char('1') => Key::Menu,
-            KeyCode::Char('2') => Key::Projects,
-            KeyCode::Char('?') => Key::Help,
-            _ => Key::Ignore,
+    /// Creates a Key instance from a given KeyEvent.
+    ///
+    /// This function takes a KeyEvent of the current key input event and returns a Key
+    /// instance. Since events where a key is double tapped can exist, a memory variable
+    /// is used. This function also returns a boolean value alongside the Key variable,
+    /// that indicates whether or not the memory should be updated.
+    ///
+    /// ## Parameters
+    /// - keyevent (crossterm::event::KeyEvent): The current key input event.
+    /// - memory (Option<KeyEvent>): The previous key input event.
+    ///
+    /// ## Returns
+    /// - A tuple containing:
+    ///   - 'Key': The created Key instance.
+    ///   - 'bool': A boolean value indicating whether to update the memory. Returns true
+    ///             if the memory should be updated and false otherwise.
+    pub fn from_keyevent(keyevent: KeyEvent, memory: Option<KeyEvent>) -> (Key, bool) {
```

Comment:
Let's change this so that the memory is a mutable reference. This can change the function return type to be a single Key and that will simplify the code a lot.